### ISSUE TITLE
aggregate `$project` mongoDB discrepancy 

### ISSUE THREAD

**Issue by saadtazi** [kofrasa/mingo#69]
I noticed a difference between mongodb and mingo behaviour: 
```
// mongoDB
db.test.insert({l1: [{l2: [{ l3: 'level3', l3bis: 'level3bis'}]}, {l2bis: 'level2bis'}]});
db.test.aggregate([{$project: { l3: '$l1.l2.l3'}}]);
// ==> { "_id" : ObjectId("..."), "l3" : [ [ "level3" ] ] }

// mingo
mingo.aggregate(
    [{_id: 'some-id', l1: [{l2: [{ l3: 'level3', l3bis: 'level3bis'}]}, {l2bis: 'level2bis'}]}],
    [{$project: { l3: '$l1.l2.l3'}}]
);
// ==> [ { _id: 'some-id', l3: 'level3' } ]
```
Notice that `l3` is not an array of array.

It is probably related to [this line](https://github.com/kofrasa/mingo/blob/master/lib/internal.js#L204).

I don't consider this a problematic issue, I just wanted to let you know...

**Comment by kofrasa** [comment]
Thanks. 

I will mark it as a bug for now.

There a few edge cases in `MongoDB` that `mingo` doesn't cover. The lack of a spec means we are left with reverse engineering by comparing outputs of many examples. 

Please what version of `MongoDB` did you test with?

**Comment by saadtazi** [comment]
mongodb v3.4.6 (the official mongo:3 [Docker image](https://hub.docker.com/r/_/mongo/))

**Comment by kofrasa** [comment]
Fixed since `v2.0.2` in https://github.com/kofrasa/mingo/commit/bc2107f6ec01c6b01b7e1f99ea6e915b0d0581c5
